src/openvic-simulation/map/Map.cpp

```diff
@@ -670,3 +670,82 @@ bool Map::generate_and_load_province_adjacencies(std::vector<LineObject> const&
 	);
 	return ret;
 }
+
+bool Map::load_climate_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid climate identifier - empty!");
+			return false;
+		}
+
+		Climate* cur_climate = climates.get_item_by_identifier(identifier);
+		if (cur_climate == nullptr) {
+			ModifierValue values;
+			bool ret = modifier_manager.expect_modifier_value(move_variable_callback(values))(node);
+			ret &= climates.add_item({ identifier, std::move(values) });
+			return ret;
+		} else {
+			bool ret = expect_list([this, &identifier, &cur_climate](ast::NodeCPtr list_element) -> bool {
+				std::string_view cur_province_id;
+				bool ret = expect_identifier_or_string(assign_variable_callback(cur_province_id))(list_element);
+				Province* cur_province = provinces.get_item_by_identifier(cur_province_id);
+				if (cur_province != nullptr) {
+					cur_climate->add_province(cur_province);
+					return ret;
+				} else {
+					Logger::error("Invalid province id ", cur_province_id, " in definition of climate ", identifier, "!");
+					return false;
+				}
+			})(node);
+			return ret;
+		}
+	})(root);
+
+	lock_climates();
+
+	for (Climate const& cl : climates.get_items()) {
+		for(Province* prov : cl.get_provinces()) {
+			prov->climate = &cl;
+		}
+	}
+
+	return ret;
+}
+
+bool Map::load_continent_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid continent identifier - empty!");
+			return false;
+		}
+
+		ModifierValue values;
+		ProvinceSetModifier::provinces_t prov;
+		bool ret = modifier_manager.expect_modifier_value_and_keys(move_variable_callback(values),
+			"provinces", ONE_EXACTLY, expect_list([this, &identifier, &prov](ast::NodeCPtr list_element) -> bool {
+				std::string_view cur_province_id;
+				bool ret = expect_identifier_or_string(assign_variable_callback(cur_province_id))(list_element);
+				Province* cur_province = provinces.get_item_by_identifier(cur_province_id);
+				if (cur_province != nullptr) {
+					prov.emplace_back(cur_province);
+					return ret;
+				} else {
+					Logger::error("Invalid province id ", cur_province_id, " in definition of continent ", identifier, "!");
+					return false;
+				}
+			})
+		)(node);
+		ret &= continents.add_item({ identifier, std::move(values), std::move(prov) });
+		return ret;
```

Comment:
The `expect_list` here can use `expect_province_identifier` similar to what I suggested for the climate loading function, the only difference is that here you don't already have a `Continent` object to `add_province` to, but with the changes in #109, you can just construct a `Continent` once you've got the `ModifierValue` loaded and use `add_provinces` on it at the end, so line 738 here becomes:
```c++
		Continent continent { identifier, std::move(values) };
		ret &= continent.add_provinces(prov);
		ret &= continents.add_item(std::move(continent));		
```

There are no duplicate provinces across continents like there are for climates in vanilla, probably because continents are fixed geographical things whereas climates are fuzzier and get changed over time. This should still use a `std::set<Province const*>` to check for duplicates across different continents and not add them if they already exist in another continent, but IMO it's fine to let the first continent a province is defined in be the one that matters, and ignore later ones, so no reverse parsing needed.